### DISCUSSION TITLE
How to get currently focused screen

### DISCUSSION THREAD

**Discussion by vdegenne** [WayfireWM/wayfire#2388] [Q&A]
Hello,

I have two screens, I need to make a script and in that script I have to determine which screen is having the focus, I believe there is a native way because `wayfire` has the `<modifier>`+`o` keybind to alterne between spaces, I just can't find a way to get this information programmatically.

Help well appreciated.

**Comment by ammen99** [comment]
ipc-rules has this: https://github.com/WayfireWM/wayfire/blob/master/plugins/ipc-rules/ipc-rules.cpp#L36 :)

**Reply by vdegenne** [reply]
Thanks! Do you know if there is a command line tool (e.g. someone that translated most of ipc instructions into a command line utility) similar to `swaymsg` for `sway` (per se `wfmsg`) that can be used to ease up writing scripts around wayfire using bash scripts directly.
If not that could be a fun little project.

**Reply by ammen99** [reply]
There isn't, at least not yet. We are [working on making a python package](https://github.com/killown/waypy) with more easy-to-use-bindings but it is still wip, some people were working on rust bindings afaik. If you make something CLI, feel free to share :)

**Reply by vdegenne** [reply]
@ammen99 I tried this simple script
```python
import wayfire_socket as ws
import os
import json

addr = os.getenv('WAYFIRE_SOCKET')
sock = ws.WayfireSocket(addr)

query = ws.get_msg_template("window-rules/get-focused-output")
response = sock.send_json(query)
print(response)
```

But I get a `"No such method found!"` error (other window rules methods work).
Is it because `get-focused-output` method was added after last release? (I'm on wayfire `0.8.1-unknown` though) ?

**Reply by ammen99** [reply]
Yes it is only on latest git.

**Reply by vdegenne** [reply]
https://github.com/WayfireWM/wf-install/issues/41
